corehq/apps/app_manager/fixtures/mobile_ucr.py

```diff
@@ -54,8 +54,23 @@ def __call__(self, restore_state):
         if not toggles.MOBILE_UCR.enabled(restore_user.domain) or not _should_sync(restore_state):
             return []
 
-        app = restore_state.params.app
-        apps = [app] if app else [a for a in get_apps_in_domain(restore_user.domain, include_remote=False)]
+        app_aware_sync_app = restore_state.params.app
+        if app_aware_sync_app:
+            apps = [app_aware_sync_app]
+        elif restore_state.params.device_id and "WebAppsLogin" in restore_state.params.device_id:
+            # Only sync reports for apps the user has access to if this is a restore from webapps
```

Comment:
Because the app visibility permission is only for webapps. 
The phone already does "app aware sync".
The fallback case is if you are syncing from the admin restore page or similar.